### ISSUE TITLE
[arrow] Build error

### ISSUE THREAD

**Issue by evgc01** [microsoft/vcpkg#22613]
**Host Environment**
- OS: Windows 10 Home (64 bits) Version 21H1 compilation 19043.1466
- Compiler: Microsoft Visual C++ 2022   00482-90000-00000-AA257

**To Reproduce**
`./vcpkg install arrow`

**Additional context**

package: arrow[core,csv,filesystem,json,parquet]:x64-windows -> 6.0.1#1
    vcpkg-tool version: 2021-12-09-724cea8b33cbf06c645f5095fa29773697da9761
    vcpkg-scripts version: bba8794b8 2022-01-14 (3 days ago)

_[---]_

Building package snappy[core]:x64-windows...
-- Downloading https://github.com/google/snappy/archive/1.1.9.tar.gz -> google-snappy-1.1.9.tar.gz...
-- Extracting source C:/dev/vcpkg/vcpkg/downloads/google-snappy-1.1.9.tar.gz
-- Using source at C:/dev/vcpkg/vcpkg/buildtrees/snappy/src/1.1.9-3644d789ea.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/dev/vcpkg/vcpkg/packages/snappy_x64-windows/share/snappy/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\_someuser_\AppData\Local\vcpkg\archives\05\0553b5e96d59dd6babb4b6b7102230861794f15ca3a504cc1aa7db4f1ee54eb6.zip
Installing package snappy[core]:x64-windows...
Elapsed time for package snappy:x64-windows: 43.19 s

_[----]_

CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:78 (message):

  C:/dev/vcpkg/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8/mingw32/bin/pkg-config.exe
  --exists arrow-compute failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "C:/dev/vcpkg/vcpkg/packages/arrow_x64-windows/lib/pkgconfig;C:/dev/vcpkg/vcpkg/packages/arrow_x64-windows/share/pkgconfig;C:/dev/vcpkg/vcpkg/installed/x64-windows/lib/pkgconfig;C:/dev/vcpkg/vcpkg/installed/x64-windows/share/pkgconfig"
      output: Package snappy was not found in the pkg-config search path.

  Perhaps you should add the directory containing `snappy.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'snappy', required by 'arrow', not found
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:201 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/arrow/portfile.cmake:112 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:142 (include)

Error: Building package arrow:x64-windows failed with: BUILD_FAILED

NOTE: I successfully installed and ran apache arrow using both MSYS2 / pacman and conda. By the way, version of snappy used by the MSYS2 package is the same as the one seemingly ignored by vcpkg (1.1.9).

Best regards.

**Comment by JackBoosY** [comment]
Please git pull vcpkg, remove  and re-install snappy then try again.

Thanks.

**Comment by JackBoosY** [comment]
Why close this? Is this resolved?

**Comment by evgc01** [comment]
Hi, Jack.

The issue is fixed (sort of), as I described in my previous post.

Essentially, the problem was two dependencies (snappy and utf8proc) were
incorrectly installed by vckpg. Two descriptor files (libutf8proc.pc
and snappy.pc)
were absent.

I just compared the source packages installed by vcpkg with the ones
installed by MSYS2 , and found the differences. Then I copied those .pc
files, and modified them to work with vcpkg.

It's more of a workaround than a fix... but it worked fine.

Best regards.
